starlark/value.go

```diff
@@ -1165,6 +1185,84 @@ func (s *Set) Union(iter Iterator) (Value, error) {
 	return set, nil
 }
 
+func (s *Set) Difference(other Iterator) (Value, error) {
+	diff := new(Set)
+	for e := s.ht.head; e != nil; e = e.next {
+		diff.Insert(e.key) // can't fail
+	}
+	var x Value
+	for other.Next(&x) {
+		if _, err := diff.Delete(x); err != nil {
+			return nil, err
+		}
+	}
+	return diff, nil
+}
+
+func (s *Set) isSuperset(other Iterator) (bool, error) {
```

Comment:
The issub/superset checks could benefit from two flavors, one for iterators and one for sets. The set-based one can perform a fast check based on sizes. I'm not sure yet which one should be exposed in the Go public API.

Reply:
I think it makes sense to expose the iterator-based one, for consistency with other methods.

Not sure if its worth adding a separate method, for setIsSuperset, but we can at least add the fast-check into compareSameTypes.